test/com/puppetlabs/http/client/impl/metrics_unit_test.clj

```diff
@@ -0,0 +1,150 @@
+      (let [java-data (Metrics/getClientMetricsDataWithUrl registry url2 bytes-read)
+            clj-data (async/get-client-metrics-data
+                      registry {:url url2 :metric-type "bytes-read"})]
+        (is (= (add-metric-ns "with-url.http://test.com/one/two.bytes-read")
+               (first (keys java-data))
+               (first (keys clj-data))))
+        (is (= 1 (.getCount (first (vals java-data)))
+               (:count (first (vals clj-data))))))
+      (testing "getClientMetricsData with url returns nothing if url is not a full match"
+        (is (= {} (Metrics/getClientMetricsDataWithUrl registry "http://test.com" bytes-read)
+               (async/get-client-metrics-data
+                registry {:url "http://test.com" :metric-type "bytes-read"})))))
+    (testing "getClientMetricsData with url and method returns the right thing"
+      (let [java-data (Metrics/getClientMetricsDataWithUrlAndVerb registry url "GET" bytes-read)
+            clj-data (async/get-client-metrics-data
+                      registry {:url url :verb "GET" :metric-type "bytes-read"})]
+        (is (= (add-metric-ns "with-url.http://test.com/one.GET.bytes-read")
+               (first (keys clj-data))
+               (first (keys java-data))))
+        (is (= 1 (.getCount (first (vals java-data)))
+               (:count (first (vals clj-data))))))
+      (let [java-data (Metrics/getClientMetricsDataWithUrlAndVerb registry url "POST" bytes-read)
+            clj-data (async/get-client-metrics-data
+                      registry {:url url :verb "POST" :metric-type "bytes-read"})]
+        (is (= (add-metric-ns "with-url.http://test.com/one.POST.bytes-read")
+               (first (keys java-data))
+               (first (keys clj-data))))
+        (is (= 2 (.getCount (first (vals java-data)))
+               (:count (first (vals clj-data))))))
+      (let [java-data (Metrics/getClientMetricsDataWithUrlAndVerb registry url2 "GET" bytes-read)
+            clj-data (async/get-client-metrics-data
+                      registry {:url url2 :verb "GET" :metric-type "bytes-read"})]
+        (is (= (add-metric-ns "with-url.http://test.com/one/two.GET.bytes-read")
+               (first (keys java-data))
+               (first (keys clj-data))))
+        (is (= 1 (.getCount (first (vals java-data)))
+               (:count (first (vals clj-data))))))
+      (testing "getClientMetricsData with url and verb returns nothing if verb is not a match"
+        (is (= {} (Metrics/getClientMetricsDataWithUrlAndVerb
+                   registry "http://test.com" "PUT" bytes-read)
+               (async/get-client-metrics-data
+                registry {:url "http://test.com" :verb "PUT" :metric-type "bytes-read"})))))
+    (testing "getClientMetricsData with metric id returns the right thing"
+      (let [java-data (Metrics/getClientMetricsDataWithMetricId
+                  registry (into-array ["foo"]) bytes-read)
```

Comment:
Nit - the word `registry` on this - and a few of the lines in other assertions below - seems misaligned per the line above it.